assignments/array-methods.js

```diff
@@ -56,28 +56,66 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+
+let upperName = runners.map((name)=>{
+    allCaps.push(`${name.first_name.toUpperCase()}`);
+});
 console.log(allCaps); 
 
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
-console.log(largeShirts);
 
+let runningLarge = runners.filter( (runner)=>{
+  if(runner['shirt_size'] === "L"){
+    largeShirts.push(runner.first_name);
+  } 
+});
+
+console.log(largeShirts);
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
+
 let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
 
+let compileDonations = runners.reduce( (ticketPriceTotal, index)=>{
+    return ticketPriceTotal + index.donation;
+},0);
+console.log(compileDonations);
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 return an array of all emails of runners
+let emails = [];
+let emailList = runners.forEach( (runner)=>{
+  return emails.push(runner.email);
+});
+console.log(emails)
+
+// Problem 2 Return an array of runners that donated more than 100 dollars
+let donation = [];
+let compileData = runners.filter( (runner)=>{
+    return donation.push(runner.donation > 100);
```

Comment:
you don't need to push to donation here because filter returns an array by default, and that array is then saved into `compileData`, so you can just console.log compileData.